starknet_py/cairo/serialization/data_serializers/payload_serializer.py

```diff
@@ -0,0 +1,85 @@
+from collections import OrderedDict
+from dataclasses import dataclass, InitVar, field
+from typing import Dict, Generator
+
+from starknet_py.cairo.serialization._context import (
+    SerializationContext,
+    DeserializationContext,
+)
+from starknet_py.cairo.serialization.data_serializers._common import (
+    deserialize_to_dict,
+    serialize_from_dict_by_key,
+    serialize_from_dict,
+)
+from starknet_py.cairo.serialization.data_serializers.array_serializer import (
+    ArraySerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.cairo_data_serializer import (
+    CairoDataSerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.felt_serializer import (
+    FeltSerializer,
+)
+
+from starknet_py.cairo.serialization.serialized_payload import SerializedPayload
+
+from starknet_py.utils.tuple_dataclass import TupleDataclass
+
+
+@dataclass
+class PayloadSerializer(CairoDataSerializer[Dict, TupleDataclass]):
+    """
+    Serializer for payloads like function arguments/function outputs/events.
+    Can serialize a dictionary.
+    Deserializes data to a TupleDataclass.
+    """
+
+    # Value present only in constructor
+    input_serializers: InitVar[OrderedDict[str, CairoDataSerializer]]
```

Comment:
Do we need this here? And if yes, tell me why 😅

Reply:
I don't want to mutate the dictionary I accept in constructor. Will add comment.